docs/the-new-architecture/turbo-modules-with-swift.md

```diff
@@ -0,0 +1,167 @@
+
+1. Open the `RCTNativeLocalStorage.mm` file
+2. Update it as it follows:
+
+```diff title="RCTNativeLocalStorage.mm"
+//  RCTNativeLocalStorage.m
+//  TurboModuleExample
+
+#import "RCTNativeLocalStorage.h"
++#import "SampleApp-Swift.h"
+
+- static NSString *const RCTNativeLocalStorageKey = @"local-storage";
+
+-@interface RCTNativeLocalStorage()
+-@property (strong, nonatomic) NSUserDefaults *localStorage;
+-@end
+
+-@implementation RCTNativeLocalStorage
++@implementation RCTNativeLocalStorage {
++    NativeLocalStorage *storage;
++}
+
+-RCT_EXPORT_MODULE(NativeLocalStorage)
+
+ - (id) init {
+   if (self = [super init]) {
+-    _localStorage = [[NSUserDefaults alloc] initWithSuiteName:RCTNativeLocalStorageKey];
++    storage = [NativeLocalStorage new];
+   }
+   return self;
+ }
+
+ - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const facebook::react::ObjCTurboModule::InitParams &)params {
+   return std::make_shared<facebook::react::NativeLocalStorageSpecJSI>(params);
+ }
+
+ - (NSString * _Nullable)getItem:(NSString *)key {
+-   return [self.localStorage stringForKey:key];
++   return [storage getItemFor:key];
+ }
+
+ - (void)setItem:(NSString *)value key:(NSString *)key {
+-   [self.localStorage setObject:value forKey:key];
++   [storage setItemFor:key value:value];
+ }
+
+ - (void)removeItem:(NSString *)key {
+-   [self.localStorage removeObjectForKey:key];
++   [storage removeItemFor:key];
+ }
+
+ - (void)clear {
+-   NSDictionary *keys = [self.localStorage dictionaryRepresentation];
+-   for (NSString *key in keys) {
+-     [self removeItem:key];
+-   }
++  [storage clear];
+ }
+
+++ (NSString *)moduleName
++{
++  return @"NativeLocalStorage";
++}
+
+@end
+```
+
+The code is not really changed. Instead of creating a reference to the `NSUserDefaults` directly, we create a new `NativeLocalStorage` from ur swift implementation and, whenever a native module function is invoked, we forward that invocation to the `NativeLocalStorage` implemented in Swift.
+
+Remember to import the `"SampleApp-Swift.h"` header. This is a header automatically generated by Xcode which contains the public API of your Swift files, in a format that is consumable by Objective-C. The `SampleApp` part of the header is actually your App name, so if you created the app with a name that is **different** from `SampleApp`, you'll have to change it.
+
+This approach introduces a bit of code duplication in the interfaces, but it allows you to reuse the Swift code you may already have in your codebase, with little extra effort.
+
+### Implementing the Bridging Header
+
+The last required step to connect the Swift code with the Objective-C++ counterpart is a bridging header.
+
+A bridging header is an header where you can import all the Objective-C header files that needs to be visible by your swift code.
+
+You might already have a bridging header in your codebase, but in case you haven't, you can create a new one by following these steps:
```

Comment:
I tmight be worth noting that library authors don't need a bridging header, only the end app. I remember this caused confusion a while back, and probably still does because the docs still say we need one.